docs/create-packages/native-files-in-net-packages.md

```diff
@@ -0,0 +1,165 @@
+The second package contains the .NET 5+ `ref/*` and `runtimes/*` assets as described above.
+The last package will contain the .NET Framework assets, and will not use the `ref/*` or `runtime/*` conventions.
+This is due to the build system having (partial) support for these NuGet conventions, but the runtime working sufficiently differently that it's easier to avoid it, to maximize project compatibility.
+
+The package with .NET Framework native libraries should have custom [MSBuild targets and props files](../concepts/MSBuild-props-and-targets.md), under `buildTransitive/<tfm>/<package ID>.[props|targets]`.
+For example, `buildTransitive/net472/Contoso.Native.targets`.
+To support `packages.config` projects, also have a `build/<tfm>/<package ID>.[props|targets]` which imports the `buildTransitive` file.
+This MSBuild script script should copy the native assemblies to subdirectories.
+
+Finally, the managed library that uses P/Invoke to call into the native library, it should have runtime checks to determine what platform the current executable is running in (x86, x64, or ARM64), and then P/Invoke into the native assembly in a correct directory.
+This is because .NET Framework supports the concept of "Any CPU" executables, which will execute (JIT compile) as x86 on x86 processors, and x64 otherwise (x64 or ARM64 processors).
+.NET Core removed this "Any CPU" executable concept, and separate x64, x86, and ARM64 executables need to be compiled and deployed explicitly.
+Therefore, managed libraries with P/Invoke do not need this runtime check on .NET Core runtimes, but do on .NET Framework runtimes.
+
+If the .NET Framework project consuming the library with managed concents explicitly opts out of "Any CPU", then these complications are not relevant, and the package author can use `runtimes/win-x64/lib/net48/contoso.dll`, for example.
+However, this is only feasible if the package is private and only used in projects that the package author controls, or at least has sufficient influence over.
+
+### SDK style projects targeting .NET Framework
+
+If you test an SDK style project that targets .NET Framework with the `runtimes/*` convention, using .NET Framework TFM, you will see that the .NET SDK will copy the x86 assets when the build or publish does not specific a RID.
+However, as described above, if the x86 `.exe` is running on an x64 or ARM64 CPU, the runtime will JIT compile x64 instructions, and the runtime will fail to execute the x86 native assembly at the moment the managed library first P/Invokes the native library.
+
+However, if `dotnet build -r win-x64` or `dotnet build -r win-arm64` (or the `publish` equivalents) are used, then the executable will be platform specific, failing to run on different CPU architectures, and the build will copy the correct native libraries.
```

Comment:
@dsplaisted @baronfel this snippet might not entirely make sense without the context of having read the document up to this point, but could you provide any feedback on this section specifically, hopefully being .NET SDK experts? I don't know how familiar you are with the .NET Framework runtime, and issues package authors have creating packages that P/Invoke into native libraries.

The bottom third of the document explain issues with all the different .NET Framework project types, so if you have knowledge in any of those areas, any feedback would be greatly appreciated. After too many hours of manual testing, to guesstimate how things work, then too many hours of writing unintelligible gibberish documentation, this is my final attempt before refusing to spend even more time without other people's help.

Clearly, documentation writing is neither something that brings me joy, not something I'm skillful at.

Reply:
Basically, if you haven't set a `RuntimeIdentifier` for a .NET Framework project, the .NET SDK will choose one for you to try to get any native assets from NuGet packages you reference.  If the `PlatformTarget` is `x64` then it will use `win7-x64` for the `RuntimeIdentifier`, otherwise if the `PlatformTarget` is `x86` or is not set, it will use `win7-x86` for the `RuntimeIdentifier`.

Once the `RuntimeIdentifier` is set, I think the rest of the logic is the same as for other builds with a `RuntimeIdentifier` specified - the native assets matching the `RuntimeIdentifier` will be copied to the output directory.

It will also set the `PlatformTarget` to match the `RuntimeIdentifier` if the `PlatformTarget` wasn't set.  Then, once it's processed the NuGet assets file, if neither the `PlatformTarget` nor the `RuntimeIdentifier` were originally set, and there were no native assets found from NuGet packages, then it will set the `PlatformTarget` back to `AnyCPU`.

That means that if you build an app targeting .NET Framework without setting the platform or RuntimeIdentifier, it will build as AnyCPU.  If you reference a NuGet package that has native assets, it will switch to x86 and use the x86 native assets from a NuGet package.  Because the `PlatformTarget` has switched to `x86`, the app will normally run successfully on an x64 machine too (while an AnyCPU app would run in x64 mode by default and would fail when trying to use x86 native assets).

The logic is described in the comment at the beginning of this file: https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets  I don't think we have it documented anywhere else so it would be great to cover in this document.

Note that in alignment with the simplified RID graph, we may change the defaults to `win-x86` and `win-x64` instead of `win7-x86` and `win7-x64`.

Reply:
@dsplaisted thank you for the information!  Sorry for not acting on the information for 3 weeks. I've just pushed some changes if you've like to review that what I wrote is still accurate and makes sense: https://github.com/NuGet/docs.microsoft.com-nuget/pull/3092/commits/37cde763c04ed155a37d3fdc9235be4039a99097